tools/firmware-utils/src/dlink-sge-image.c

```diff
@@ -0,0 +1,415 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later OR MIT */
+/*
+ * This tool encrypts factory images for certain D-Link Devices
+ * manufactured by SGE / T&W, e.g. COVR-C1200, COVR-P2500, DIR-882, ...
+ *
+ * Build instructions:
+ *   gcc -lcrypto dlink-sge-image.c -o dlink-sge-image
+ *
+ * Usage:
+ *   ./dlink-sge-image infile outfile [-d: decrypt]
+ *
+ * Copyright (C) 2021 Sebastian Schaper <openwrt@sebastianschaper.net>
```

Comment:
I'm not sure, but I'd have moved the Copyright to the top of the comment, so it's closest to the SPDX identifier.